stdlib/http/src/main/ballerina/src/http/client_endpoint.bal

```diff
@@ -204,7 +275,14 @@ public type Client client object {
     # + promise - The related `PushPromise`
     # + return - A promised HTTP `Response` message, or an error if the invocation fails
     public remote function getPromisedResponse(PushPromise promise) returns Response|ClientError {
-        return self.httpClient->getPromisedResponse(promise);
+        Response|ClientError response = self.httpClient->getPromisedResponse(promise);
+        if (response is Response) {
+            error? x = observe:addTagToSpan(HTTP_STATUS_CODE_GROUP, getStatusCodeRange(response.statusCode));
```

Comment:
We can add the path and method from the push promise can't we ?

Reply:
Fixed with 527d3559b6c04d5002739eaffbabbfa83ea6856a